common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisFlightEventManager.java

```diff
@@ -0,0 +1,276 @@
+    }
+
+    public CompoundTag saveData(CompoundTag tag) {
+        tag.putBoolean("isWaitingForControlResponse", this.isWaitingForControlResponse);
+        tag.putBoolean("isInDangerZone", this.isInDangerZone);
+        tag.putInt("ticksInTheDangerZone", this.ticksInTheDangerZone);
+        tag.putInt("requiredDangerZoneRequests", this.requiredControlRequests);
+        tag.putInt("dangerZoneResponses", this.dangerZoneResponses);
+        tag.putInt("controlRequestCooldown", this.controlRequestCooldown);
+        tag.putInt("ticksSincePrompted", this.ticksSincePrompted);
+        tag.putFloat("dangerZoneShakeScale", this.dangerZoneShakeScale);
+
+        if (this.controlPrompt != null) {
+            tag.putString("controlPrompt", this.controlPrompt.getSerializedName());
+        }
+
+        return tag;
+    }
+
+    public boolean isWaitingForControlResponse() {return isWaitingForControlResponse;}
+    public ConsoleControl getWaitingControlPrompt() {return this.controlPrompt;}
+    public int getControlResponses() {return this.controlResponses;}
+    public int getRequiredControlRequests() {return this.requiredControlRequests;}
+
+    public boolean isInDangerZone() {
+        return isInDangerZone;
+    }
+    public float dangerZoneShakeScale() {
+        return this.dangerZoneShakeScale;
+    }
+
+    public boolean areControlEventsComplete() {
+        return this.controlResponses == this.requiredControlRequests;
+    }
+
+    public boolean areDangerZoneEventsComplete() {
+        return this.requiredDangerZoneRequests <= this.dangerZoneResponses;
+    }
+
+    // "Combo Time" refers to the prime time between choosing controls
+    private boolean isEventInComboTime() {
+        return (this.ticksSincePrompted < 3 * 20);
+    }
+
+    private int getControlRequestCooldown() {
+        return (isEventInComboTime() ? 20 : 60);  // This will be expanded on when Stats are added.
+    }
+
+
+    public void calculateTravelLogic() {
+
+        // Only trigger a responses reset if we haven't started flight yet.
+        if (this.requiredControlRequests == 0) {
+            this.controlResponses = 0;
+            this.isWaitingForControlResponse = false;
+            this.controlRequestCooldown = getControlRequestCooldown();
+        }
+
+        // Calculate the distance between two points
+        var current = this.operator.getExteriorManager().getLastKnownLocation().position;
+        var target = this.operator.getControlManager().getTargetLocation().position;
+        Vec3 currentVec = new Vec3(current.getX(), current.getY(), current.getZ());
+        Vec3 targetVec = new Vec3(target.getX(), target.getY(), target.getZ());
+
+        var distance = currentVec.distanceTo(targetVec);
+
+        // Determine if the distance is worth the prompts
+        if (distance > MIN_DISTANCE_FOR_EVENTS) {
+            this.requiredControlRequests = getBlocksPerRequest(distance);
+        } else {
+            this.requiredControlRequests = 0;
```

Comment:
Is this check needed? It looks like ``requiredControlRequests`` doesn't get modified in other instances. If that is indeed the case, perhaps this check is not needed. We can define the initial value to be 0 in the field declaration.

Reply:
Resolved in upcoming push